modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1900,13 +1900,36 @@ def str_split(self, pat=None, n=-1, expand=False, regex=None):
 
     # END String map partitions operations
 
-    def unique(self):
-        new_modin_frame = self._modin_frame.apply_full_axis(
-            0,
-            lambda x: x.squeeze(axis=1).unique(),
-            new_columns=self.columns,
+    def unique(self, keep="first", ignore_index=True, subset=None):
+        # kernels with 'pandas.Series.unique()' work faster
+        can_use_unique_kernel = (
+            subset is None and ignore_index and len(self.columns) == 1 and keep
         )
-        return self.__constructor__(new_modin_frame)
+
+        if not can_use_unique_kernel and not RangePartitioning.get():
+            return super().unique(keep=keep, ignore_index=ignore_index, subset=subset)
```

Comment:
Is this branch for d2p?

Reply:
no, this branch is for [general `unique()` implementation](https://github.com/dchigarev/modin/blob/df12976e895f4840c6368e5b3350e1f007dc9fb7/modin/core/storage_formats/base/query_compiler.py#L1790-L1797) that uses QC's public API